packages/vis/src/composables/useTimeline.ts

```diff
@@ -0,0 +1,348 @@
+import { 
+  computed, 
+  onMounted, 
+  onUnmounted, 
+  ref, 
+  watch 
+} from 'vue-demi'
+
+import type { Ref } from 'vue-demi'
+
+import { MaybeElementRef } from '@vueuse/core'
+
+import { resolveUnref } from '@vueuse/core'
+
+import type { 
+  DataItem, 
+  DataGroup, 
+  TimelineOptions, 
+  TimelineAnimationOptions, 
+  TimelineEventPropertiesResult 
+} from "vis-timeline/esnext"
+
+import { Timeline } from "vis-timeline/esnext"
+
+export interface UseTimelineParams {
+  /**
+   * 
+   * 
+   * Items for the Timeline.
+   * 
+   */
+  items: DataItem[]
+  /**
+   * 
+   * 
+   * Groups for the Timeline.
+   * 
+   */
+  groups?: DataGroup[]
+  /**
+   * 
+   * 
+   * 
+   * 
+   */
+  currentTime?: Ref<Date | string>
+  /**
+   * 
+   * 
+   * Timeline selections
+   * 
+   */
+   selection?: Ref<string[] | string>
+  /**
+   * 
+   * 
+   * Timeline options.
+   * 
+   */
+  options?: TimelineOptions
+}
+
+export const useTimeline = (
+  el: MaybeElementRef<HTMLElement | undefined>,
```

Comment:
Why do we need to wrap the HTMLElement in a `MaybeElementRef` type? What does it do?

Reply:
So this allows us to pass in either a `ref()` of the element, or `element.value`. It's slightly confusing but gives flexibility. :)